### PR TITLE
Update comparison for GitLab CE

### LINKED ISSUE #13018: Comparison page is outdated for GitLab CE
Browsing the [comparison page](https://docs.gitea.io/en-us/comparison/), I found some outdated things for GitLab CE. Opening this issue according to the page. There might be more, but a short list:

- Related issues are supported now
- Service Desk is supported now
- Squash merging should be supported now

Let me know if you need references for those.

### PR DESCRIPTION
**Pull request by claell**
Fixes #13018.

### CODE REVIEW

File: docs/content/doc/features/comparison.en-us.md [hunk 1]
```diff
@@ -79,7 +79,7 @@ _Symbols used in table:_
 | Labels | ✓ | ✓ | ✓ | ✓ | ✓ | ✘ | ✘ |
 | Time tracking | ✓ | ✘ | ✓ | ✓ | ✓ | ✘ | ✘ |
 | Multiple assignees for issues | ✓ | ✘ | ✓ | ✘ | ✓ | ✘ | ✘ |
-| Related issues | ✘ | ✘ | ⁄ | ✘ | ✓ | ✘ | ✘ |
+| Related issues | ✘ | ✘ | ⁄ | [✓](https://docs.gitlab.com/ee/user/project/issues/related_issues.html#related-issues) | ✓ | ✘ | ✘ |
```
Comment (mrsdizzie) [inline docs/content/doc/features/comparison.en-us.md]:
```suggestion
| Related issues | ✘ | ✘ | ⁄ | [✓](https://docs.gitlab.com/ee/user/project/issues/related_issues.html) | ✓ | ✘ | ✘ |
```

File: docs/content/doc/features/comparison.en-us.md [hunk 2]
```diff
@@ -79,7 +79,7 @@ _Symbols used in table:_
 | Labels | ✓ | ✓ | ✓ | ✓ | ✓ | ✘ | ✘ |
 | Time tracking | ✓ | ✘ | ✓ | ✓ | ✓ | ✘ | ✘ |
 | Multiple assignees for issues | ✓ | ✘ | ✓ | ✘ | ✓ | ✘ | ✘ |
-| Related issues | ✘ | ✘ | ⁄ | ✘ | ✓ | ✘ | ✘ |
+| Related issues | ✘ | ✘ | ⁄ | [✓](https://docs.gitlab.com/ee/user/project/issues/related_issues.html) | ✓ | ✘ | ✘ |
```
Comment (techknowlogick) [inline docs/content/doc/features/comparison.en-us.md]:
```suggestion
| Related issues | ✘ | ✘ | ⁄ | [✓](https://docs.gitlab.com/ce/user/project/issues/related_issues.html) | ✓ | ✘ | ✘ |
```

File: docs/content/doc/features/comparison.en-us.md [hunk 3]
```diff
@@ -90,14 +90,14 @@ _Symbols used in table:_
...
-| Service Desk | [✘](https://github.com/go-gitea/gitea/issues/6219) | ✘ | ✘ | ✘ | ✓ | ✘ | ✘ |
+| Service Desk | [✘](https://github.com/go-gitea/gitea/issues/6219) | ✘ | ✘ | [✓](https://gitlab.com/groups/gitlab-org/-/epics/3103) | ✓ | ✘ | ✘ |
 
 #### Pull/Merge requests
 
 | Feature | Gitea | Gogs | GitHub EE | GitLab CE | GitLab EE | BitBucket | RhodeCode CE |
 |---------|-------|------|-----------|-----------|-----------|-----------|--------------|
 | Pull/Merge requests | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ | ✓ |
-| Squash merging | ✓ | ✘ | ✓ | ✘ | ✓ | ✓ | ✓ |
+| Squash merging | ✓ | ✘ | ✓ | [✓](https://docs.gitlab.com/ee/user/project/merge_requests/squash_and_merge.html) | ✓ | ✓ | ✓ |
```
Comment (techknowlogick) [inline docs/content/doc/features/comparison.en-us.md]:
```suggestion
| Squash merging | ✓ | ✘ | ✓ | [✓](https://docs.gitlab.com/ce/user/project/merge_requests/squash_and_merge.html) | ✓ | ✓ | ✓ |
```

**Review by mrsdizzie — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]
Thanks for such a quick turnaround :)

**Review by lafriks — APPROVED** [review]

**Comment by claell** [comment]
Thansk for the fast review!

**Pull request merged**